components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAAdminService.java

```
@@ -0,0 +1,288 @@
+        } catch (CAException e) {
+            log.error("Error listing certificates for tenant:" + tenantDomain, e);
+            throw new CAException("Error listing the certificates");
+        }
+    }
+
+    /**
+     * Lists all the certificates issued by the tenant CA filtered by the given status.
+     *
+     * @param status The status filter
+     * @return List of certificates with given status issued by tenant CA
+     * @throws CAException
+     * @see org.wso2.carbon.identity.certificateauthority.common.CertificateStatus
+     */
+    public Certificate[] listCertificatesWithStatus(String status) throws CAException {
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            List<Certificate> certificateInfoList = certificateService.listCertificates(status, tenantDomain);
+            return certificateInfoList.toArray(new Certificate[certificateInfoList.size()]);
+        } catch (CAException e) {
+            log.error("Error listing certificates with status " + status + " for tenant:" + tenantDomain, e);
+            throw new CAException("Error listing \"" + status + "\" certificates");
+        }
+    }
+
+    /**
+     * Get the details of the certificate identified by the given serial number.
+     *
+     * @param serialNo The serial number of the certificate
+     * @return The certificate
+     * @throws CAException
+     */
+    public Certificate getCertificate(String serialNo) throws CAException {
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            return certificateService.getCertificate(serialNo, tenantDomain);
+        } catch (CAException e) {
+            log.error("Error retrieving the certificate with serial no:" + serialNo + " from tenant:" + tenantDomain,
+                    e);
+            throw new CAException("Error when retrieving the certificate");
```

The original exception is swallowed here. Please fix.